src/server/repositories/UrlRepository.ts

```diff
@@ -167,6 +173,186 @@ export class UrlRepository implements UrlRepositoryInterface {
     }
   }
 
+  public rawDirectorySearch: (
+    query: string,
+    order: SearchResultsSortOrder,
+    limit: number,
+    offset: number,
+    state: string | undefined,
+    isFile: boolean | undefined,
+    isEmail: boolean,
+  ) => Promise<UrlDirectoryPaginated> = async (
+    query,
+    order,
+    limit,
+    offset,
+    state,
+    isFile,
+    isEmail,
+  ) => {
+    const { tableName } = Url
+
+    const urlVector = urlSearchVector
+
+    const rankingAlgorithm = this.getRankingAlgorithm(order, tableName)
+
+    if (isEmail) {
+      const emails = query.toString().split(' ')
+
+      // split email/domains by space into tokens, also reduces injections
+      const likeQuery: Array<string> = []
+      emails.forEach((domain) => {
+        likeQuery.push(sanitise(domain))
+      })
+
+      const urlsModel = await this.getRelevantUrlsFromEmail(
+        likeQuery,
+        rankingAlgorithm,
+        limit,
+        offset,
+        state,
+        isFile,
+      )
+
+      return urlsModel
+    }
+
+    const urlsModel = await this.getRelevantUrlsFromText(
+      urlVector,
+      rankingAlgorithm,
+      limit,
+      offset,
+      query,
+      state,
+      isFile,
+    )
```

Comment:
the only reason why you have the if-block to handle emails is for the special treatment of `likeQuery`. Given that you can express that largely in one-line, consider using a tenary to make things more concise, ie
```js
const urlsModel = await (
  isEmail
    ? this.getRelevantUrlsFromEmail(...)
    : this.getRelevantUrlsFromText(...)
)
```